.github/workflows/ci.yaml

```diff
@@ -17,17 +17,22 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
+      - name: Cache Go Modules
+        uses: actions/cache@v4
         with:
-          fetch-depth: 1
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-gomod-ci-${{ hashFiles('gradle.properties', 'gradle/libs.versions.toml') }}
+          restore-keys:
+            ${{ runner.os }}-gomod-ci-
       - uses: actions/setup-java@v4
         with:
           distribution: 'temurin'
-          java-version: '17'
+          java-version: '21'
```

Comment:
Bumping to latest LTS - we still build with source/target/release set to Java 8 compatibility.